### ISSUE TITLE
Critical dependencies error

### ISSUE THREAD

**Issue by gunta** [webpack/webpack#205]
Hi!
I get a **'Critical dependencies'** error when trying to use the Promise library [Bluebird](https://github.com/petkaantonov/bluebird).

I'm not exactly sure what does this means.

```
WARNING in ./res/bower_components/bluebird/js/browser/bluebird.js
Critical dependencies.
 @ ./res/bower_components/bluebird/js/browser/bluebird.js 27:376-383

WARNING in ./res/bower_components/bluebird/js/browser/bluebird.js
Critical dependencies.
 @ ./res/bower_components/bluebird/js/browser/bluebird.js 27:642-649
```

Thank you!

**Comment by altano** [comment]
I was having trouble getting the above methods to work with knockout 3.2.  I tried just using noParse but the resulting module export was an empty object.  Looking at the knockout code, if require and define were not present, it would export a global variable and not export it.

So there were two options: use script-loader (and lose minification or switch to the .min.js flavor) or, since knockout is never actually going to call `require` in my code base, export a require function that just throws an exception if ever called.  I opted for the latter.

```javascript
var errorFn = function() { throw new Error("knockout attempting to require"); }

var base = {
    ...
    module: {
        loaders: [
            { test: /knockout.build.output.knockout-latest(\.debug)?\.js$/, loader: "imports?require=>" + errorFn.toString() }
```

Why is it so hard for people to just use UMD. =\
